privacyscore/backend/management/commands/scanfromfile.py

```diff
@@ -27,24 +27,28 @@ class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument('file_path')
+        parser.add_argument('scan_list')
         parser.add_argument('-s', '--sleep-between-scans', type=float, default=0)
 
     def handle(self, *args, **options):
         if not os.path.isfile(options['file_path']):
             raise ValueError('file does not exist!')
         self.stdout.write('Reading from file {}'.format(options['file_path']))
 
-        start = timezone.now()
+        scan_list = ScanList.objects.get_or_create(name=options['scan_list'])[0]
+        scan_list.private = True
+        scan_list.save()
 
         with open(options['file_path'], 'r') as fdes:
-            # TODO: use bulk create for db optimization?
-            sites = {
-                Site.objects.get_or_create(
-                    url=normalize_url(url))[0]
-                for url in fdes.readlines()
-                if '.' in url
-            }
+            sites = []
+            for url in fdes.readlines():
+                if '.' in url:
```

Comment:
Is branching here necessary?  I would expect the normalisation function to be idempotent for already normalised input. Or do you intend to only normalise URLs with a "."? Then I'm interested in learning why.

Reply:
normalize_url converts the urls in the 'http://' format since the original domains in the txt file are not in that format.

Reply:
the "." check prevents reading empty lines in file